### PR TITLE
Fixes disabling of clicking after crop is deleted and added test for the cropUI

### LINKED ISSUE #1476: Area selected by image-area-select.
if we delete the crop step then area selected by image-area-select.in that area we are not able to click.
Untitled_ Jan 12, 2020 11_49 PM

### PR DESCRIPTION
**Pull request by keshav234156**
Fixes #1476 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `npm test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts
* [ ] PR is descriptively titled
* [ ] ask `@publiclab/is-reviewers` for help, in a comment below
* [ ] Insert-step functionality is working correct as expected.
> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software
Please make sure to get at least two reviews before asking for merging the PR as that would make the PR more reliable on our part
Thanks!

### CODE REVIEW

**Comment by jywarren** [comment]
Hi, thank you! I wonder if this one may need a UI test as it's a delicate interaction. What do you think?

**Comment by keshav234156** [comment]
@jywarren @HarshKhandeparkar test also  added!!

File: examples/lib/defaultHtmlStepUi.js [hunk 1]
```diff
@@ -401,7 +401,9 @@ function DefaultHtmlStepUi(_sequencer, options) {
     if (_sequencer.steps.length - 1 > 1) $('#load-image .insert-step').prop('disabled', false);
     else $('#load-image .insert-step').prop('disabled', true);
 
-    $('div[class*=imgareaselect-]').remove();
+    $(step.imgElement).imgAreaSelect({
```
Comment (HarshKhandeparkar) [inline examples/lib/defaultHtmlStepUi.js]: Can we use `$step` here instead of global jquery?

File: test/ui-2/test/cropui.test.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+const timeout = 300000 ;
+const fs = require('fs');
+//const {installMouseHelper} = require('./install-mouse-helper');
+beforeAll(async () => {
+  path = fs.realpathSync('file://../examples/index.html');
+  await page.goto('file://' + path, {waitUntil: 'domcontentloaded'});
+  await page.setViewport({ width: 1500, height: 700});
+  //await installMouseHelper(page);
+});
+describe('Crop UI', ()=>{
```
Comment (HarshKhandeparkar) [inline test/ui-2/test/cropui.test.js]: Please add some line breaks and comments to make it more readable.

**Comment by keshav234156** [comment]
@jywarren @HarshKhandeparkar Please review the changes!!

File: examples/lib/defaultHtmlStepUi.js (hunk 1 above)
Comment (HarshKhandeparkar) [reply examples/lib/defaultHtmlStepUi.js]: ?

File: test/ui-2/test/cropui.test.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+const timeout = 300000 ;
+const fs = require('fs');
+beforeAll(async () => {
+  path = fs.realpathSync('file://../examples/index.html');
+  await page.goto('file://' + path, {waitUntil: 'domcontentloaded'});
+  await page.setViewport({ width: 1500, height: 700});
+});
+describe('Crop UI', ()=>{
+  test('Crop UI', async()=>{
+    //Wait for the load step to get loaded
```
Comment (HarshKhandeparkar) [inline test/ui-2/test/cropui.test.js]: Please add a space at the start of every comment. Also, add a full stop at the end of sentences. We want the comments to be professional and readable.

File: test/ui-2/test/cropui.test.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    await page.waitForSelector('.step');
+    const Length = await page.evaluate(() => document.querySelectorAll('.step').length);
+    //Click on the Crop step from radio-group
+    await page.click('[data-value=\'crop\']');
+    const LengthChanged = await page.evaluate(() => document.querySelectorAll('.step').length);
+    const src1 = await page.evaluate(() => document.querySelectorAll('.step img')[1].src);
+    const example = await page.$$('img.step-thumbnail');
+    /**
+    *Drawing a bounding box around the image to be cropped
```
Comment (HarshKhandeparkar) [inline test/ui-2/test/cropui.test.js]: Please add a space after the `*`

File: examples/lib/defaultHtmlStepUi.js [hunk 5]
```diff
@@ -403,7 +403,9 @@ function DefaultHtmlStepUi(_sequencer, options) {
...
```
Comment (HarshKhandeparkar) [inline examples/lib/defaultHtmlStepUi.js]: Okay I found something.
https://github.com/publiclab/image-sequencer/blob/423422a61f61f1a4fa9149b9dafee1508bbe350e/examples/lib/defaultHtmlStepUi.js#L83

`step.imgElement` is already a jquery object. You don't need to wrap it in another `$()`

Comment (keshav234156) [reply examples/lib/defaultHtmlStepUi.js]: we need to do it otherwise we have an error step.imgElement.imgAreaSelect is not a function

**Comment by keshav234156** [comment]
@jywarren @HarshKhandeparkar Please re-review!!

File: test/ui-2/test/cropui.test.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    const src1 = await page.evaluate(() => document.querySelectorAll('.step img')[1].src);
+    const example = await page.$$('img.step-thumbnail');
+    /**
+    * Drawing a bounding box around the image to be cropped.
+    * box.x returns x position of mouse pointer.
+    * box.y returns y position of mouse pointer.
+    **/
+    const box = await example[1].boundingBox();
+    // Moving mouse pointer inside the the image.
+    await page.mouse.move(box.x, box.y);
```
Comment (jywarren) [inline test/ui-2/test/cropui.test.js]: Wow, this is intense, and awesome!!!

**Comment by keshav234156** [comment]
buddy-screenshot
buddy-screenshot1
buddy-screenshot3
buddy-screenshot4
Some screencshorts of the test!!!

**Comment by jywarren** [comment]
Hooray!!!

**Comment by jywarren** [comment]
Gosh, we should bump the version number soon!

**Pull request merged**